src/main/java/io/github/communitymod/core/CommonEvents.java

```diff
@@ -30,60 +49,321 @@
+                            entity.getType().getDescription().getString() +
+                            " " + ColorText.RED + Math.round((entity.getHealth() + entity.getAbsorptionAmount()) * 5 * (mobLevel.getMobLevel() / 2f)) + "/" + Math.round(entity.getMaxHealth() * 5 * (mobLevel.getMobLevel() / 2f))));
+                });
+            }
+        }
+
+        @SubscribeEvent
+        public static void PlayerHeal(final LivingHealEvent event) {
+            if (event.getEntityLiving() instanceof Player player) {
+                player.getCapability(CapabilityPlayerSkills.PLAYER_STATS_CAPABILITY).ifPresent(
+                        skills -> event.setAmount((event.getAmount() / skills.getMaxHealth()) * 100));
+            }
+        }
+
+        @SubscribeEvent
+        public static void registerCommands(final RegisterCommandsEvent event) {
+            CommandInit.registerCommands(event);
+        }
+    }
+
+    @EventBusSubscriber(modid = CommunityMod.MODID, bus = Bus.MOD)
+    public static final class ModEvents {
+
+        @SubscribeEvent
+        public static void commonSetup(final FMLCommonSetupEvent event) {
+            DimensionInit.setup();
+
+            //Structures
+            register(BeanPieces.BEAN_TYPE, "bean_structure");
+            StructureInit.setupStructures();
+            ConfiguredStructures.registerConfiguredStructures();
+
+            //Capabilities
+            CapabilityPlayerSkills.register();
+            CapabilityMobLevel.register();
+        }
+
+        @SubscribeEvent
+        public static void gatherData(final GatherDataEvent event) {
+            DataGenerator generator = event.getGenerator();
+            ExistingFileHelper existingFileHelper = event.getExistingFileHelper();
+
+            if (event.includeServer()) {
+                //TODO: recipes, loot tables, tags
+            }
+            if (event.includeClient()) {
+                generator.addProvider(new ModDataGeneration.LanguageGen(generator, "en_us"));
             }
-		}
+        }
 
-		@SubscribeEvent
-		public static void entityAttributes(final EntityAttributeCreationEvent event) {
-			event.put(EntityInit.BEAN_ENTITY.get(), BeanEntity.createAttributes().build());
-		}
+        @SubscribeEvent
+        public static void entityAttributes(final EntityAttributeCreationEvent event) {
+            event.put(EntityInit.BEAN_ENTITY.get(), BeanEntity.createAttributes().build());
+        }
 
-		private static StructurePieceType register(StructurePieceType structurePiece, String key) {
-			return Registry.register(Registry.STRUCTURE_PIECE, key.toLowerCase(Locale.ROOT), structurePiece);
-		}
+        private static StructurePieceType register(StructurePieceType structurePiece, String key) {
+            return Registry.register(Registry.STRUCTURE_PIECE, key.toLowerCase(Locale.ROOT), structurePiece);
+        }
     }
-}
\ No newline at end of file
+}
```

Comment:
Since these are such massive changes, consider moving these events to their own class.

Reply:
Moved Gameplay-related events to GameplayEvents Class.